### PR TITLE
Fix iptables backend in kpng-local-up

### LINKED ISSUE #91: Fix iptables compilation in kpng-local-up
mostly working but looks like hitting some 1.23 import errors ... 
```
ubuntu@jay-buildbox-6 [18:20:03] [~/SOURCE/kpng/cmd] [fix-iptables_backend *]
-> % go install -trimpath ./...

# k8s.io/client-go/listers/policy/v1
../../../go/pkg/mod/k8s.io/client-go@v0.23.0-alpha.1/listers/policy/v1/eviction.go:33:41: undefined: "k8s.io/api/policy/v1".Eviction
../../../go/pkg/mod/k8s.io/client-go@v0.23.0-alpha.1/listers/policy/v1/eviction.go:50:65: undefined: "k8s.io/api/policy/v1".Eviction
../../../go/pkg/mod/k8s.io/client-go@v0.23.0-alpha.1/listers/policy/v1/eviction.go:52:25: undefined: "k8s.io/api/policy/v1".Eviction
../../../go/pkg/mod/k8s.io/client-go@v0.23.0-alpha.1/listers/policy/v1/eviction.go:67:41: undefined: "k8s.io/api/policy/v1".Eviction
../../../go/pkg/mod/k8s.io/client-go@v0.23.0-alpha.1/listers/policy/v1/eviction.go:70:21: undefined: "k8s.io/api/policy/v1".Eviction
../../../go/pkg/mod/k8s.io/client-go@v0.23.0-alpha.1/listers/policy/v1/eviction.go:82:73: undefined: "k8s.io/api/policy/v1".Eviction
../../../go/pkg/mod/k8s.io/client-go@v0.23.0-alpha.1/listers/policy/v1/eviction.go:84:25: undefined: "k8s.io/api/policy/v1".Eviction
../../../go/pkg/mod/k8s.io/client-go@v0.23.0-alpha.1/listers/policy/v1/eviction.go:90:53: undefined: "k8s.io/api/policy/v1".Eviction
../../../go/pkg/mod/k8s.io/client-go@v0.23.0-alpha.1/listers/policy/v1/eviction.go:98:15: undefined: "k8s.io/api/policy/v1".Eviction
# k8s.io/client-go/applyconfigurations/meta/v1
../../../go/pkg/mod/k8s.io/client-go@v0.23.0-alpha.1/applyconfigurations/meta/v1/unstructured.go:52:13: undefined: managedfields.GvkParser
../../../go/pkg/mod/k8s.io/client-go@v0.23.0-alpha.1/applyconfigurations/meta/v1/unstructured.go:58:61: undefined: managedfields.GvkParser
../../../go/pkg/mod/k8s.io/client-go@v0.23.0-alpha.1/applyconfigurations/meta/v1/unstructured.go:69:9: undefined: managedfields.NewGVKParser
../../../go/pkg/mod/k8s.io/client-go@v0.23.0-alpha.1/applyconfigurations/meta/v1/unstructured.go:128:33: too many arguments in call to managedfields.ExtractInto
        have (*unstructured.Unstructured, typed.ParseableType, string, *unstructured.Unstructured, string)
        want (runtime.Object, typed.ParseableType, string, interface {})
```

### PR DESCRIPTION
**Pull request by jayunit100**
fixes #91

### CODE REVIEW

File: backends/iptables/register.go [hunk 1]
```diff
@@ -0,0 +1,16 @@
+package iptables
+
+import (
+//        "github.com/spf13/pflag"
```
Comment (mcluseau) [inline backends/iptables/register.go:4]: commented imports should be removed

File: backends/iptables/iptables.go [hunk 2]
```diff
@@ -24,6 +25,14 @@ var (
 	OnlyOutput = flag.Bool("only-output", false, "Only output the ipvsadm-restore file instead of calling ipvsadm-restore")
 )
 
+func BindFlags(flags *pflag.FlagSet) {
+	        flags.AddFlagSet(flag)
+}
+
+func init() {
+	backendcmd.Register("to-iptables", func() backendcmd.Cmd { return &Backend{} })
```
Comment (mcluseau) [inline backends/iptables/iptables.go:33]: duplicate with register in `register.go` isn't it?

File: hack/kpng-local-up.sh [hunk 3]
```diff
@@ -4,10 +4,9 @@
 # TODO Replace with 1.22 once we address 
 #: ${KIND:="kindest/node:v1.21.1@sha256:69860bda5563ac81e3c0057d654b5253219618a22ec3a346306239bba8cfa1a6"}
 : ${KIND:="kindest/node:v1.22.0@sha256:b8bda84bb3a190e6e028b1760d277454a72267a5454b57db34437c34a588d047"}
-: ${IMAGE:="jayunit100/kpng:ipvs"}
+: ${IMAGE:="jayunit100/kpng:2"}
```
Comment (mcluseau) [inline hack/kpng-local-up.sh:7]: it think this change is not required ;)

File: hack/kpng-local-up.sh [hunk 4]
```diff
@@ -4,10 +4,9 @@
...
 : ${PULL:=IfNotPresent}
-: ${BACKEND:=nft}
-
+: ${BACKEND:=iptables}
```
Comment (mcluseau) [inline hack/kpng-local-up.sh:9]: it think this change is not required ;)

File: hack/kpng-local-up.sh [hunk 5]
```diff
@@ -34,7 +33,8 @@ function build {
 
     docker build -t $IMAGE ./
     docker push $IMAGE
-
+ 
+    pwd
```
Comment (mcluseau) [inline hack/kpng-local-up.sh:37]: it think this change is not required ;)

File: server/go.mod [hunk 6]
```diff
@@ -2,6 +2,10 @@ module sigs.k8s.io/kpng/server
 
 go 1.17
 
+replace sigs.k8s.io/kpng/client => ../client
+
+replace sigs.k8s.io/kpng/api => ../api
```
Comment (mcluseau) [inline server/go.mod:7]: merge in one replace section instead

**Review by mcluseau — CHANGES_REQUESTED** [review]

File: hack/kpng-local-up.sh (hunk 3 above)
Comment (jayunit100) [reply hack/kpng-local-up.sh:7]: i just did `kpng:2` because `:ipvs` is super confusing

File: hack/kpng-local-up.sh (hunk 4 above)
Comment (jayunit100) [reply hack/kpng-local-up.sh:9]: nft doesnt work, iptables more common so seemed like the right thing to do?

File: hack/kpng-local-up.sh (hunk 5 above)
Comment (jayunit100) [reply hack/kpng-local-up.sh:37]: ha fair , fixing

File: Dockerfile [hunk 7]
```diff
@@ -24,7 +24,7 @@ run for f in $(find -name go.mod); do d=$(dirname $f); echo "downloading mods in
 
 add . ./
 #run for f in $(find -name go.mod); do d=$(dirname $f); echo "testing in $d"; ( cd $d && go test ./... ); done
-run cd cmd && go install -trimpath ./...
+run cd cmd && pwd && go mod tidy ; go install -trimpath ./...
```
Comment (mcluseau) [inline Dockerfile:27]: ah missed this one, please revert to the original, `go mod tidy` should be ok without issuing the command again, and the `&&` is broken.

Comment (jayunit100) [reply Dockerfile:27]: Ok Let me test and make sure

**Pull request merged**